### PR TITLE
chore(deps): dependencies updates

### PR DESCRIPTION
**Pull request by abhi-markan**
## Introduction ✏️ 
Dependency updates

## Resolution ✔️ 
* Removed few GitHub notified vulnerabilities.
* Updated dependencies.

## Miscellaneous ➕ 
* Updated `.dockerignore`
* Updated `.gitignore`
* Updated GHA : Dependabot lint and test pipeline execution.

### CODE REVIEW

**Review by Zainzzkk — DISMISSED** [review]

File: src/ui/.dockerignore [hunk 1]
```diff
@@ -1,10 +1,53 @@
...
+**/generated_reports
+**/temp/*
+**/actionsheets/*
+**/json/*
+**/amendments/*
+**/exports/*
+**/bss-ewcs/dump/*
+**/bss-ewcs/data/*
+**/gef/dump/*
+**/gef/data/*
```
Comment (ttbarnes) [inline src/ui/.dockerignore:51]: BSS EWCS, GEF?

Comment (abhi-markan) [reply src/ui/.dockerignore]: Have removed.

File: src/ui/server/helpers/task-list/index.ts [hunk 2]
```diff
@@ -42,7 +42,7 @@ export const generateSimplifiedTaskList = (taskList: TaskListData): Array<TaskLi
           status: task.status,
           title: task.title,
         })),
-      } as TaskListGroup),
+      }) as TaskListGroup,
```
Comment (Zainzzkk) [inline src/ui/server/helpers/task-list/index.ts:45]: my linting normally throws an error for this if its not all in the bracket

Comment (abhi-markan) [reply src/ui/server/helpers/task-list/index.ts:45]: Not sure why as we all should have same linting. Are you using VS code linting on top of prettier?

Comment (ttbarnes) [reply src/ui/server/helpers/task-list/index.ts:45]: I think this is now enforced with new version of prettier so without an `npm install` the linting would be different?

**Review by Zainzzkk — APPROVED** [review]

**Pull request merged**